pkg/core/mpt/proof_test.go

```diff
@@ -71,3 +77,77 @@ func TestVerifyProof(t *testing.T) {
 		require.Equal(t, []byte("somevalue"), v)
 	})
 }
+
+func TestTraverse(t *testing.T) {
+	tr := newProofTrie(t, false)
+	expectedRoot := tr.StateRoot()
+
+	t.Run("Good, restrict size", func(t *testing.T) {
+		const maxSize = 300
+		var (
+			nodes    [][]byte
+			maxBytes = maxSize
+		)
+		stop := func(node []byte) bool {
+			if len(node)+io.GetVarSize(len(node)) > maxBytes {
+				return true
+			}
+			nodes = append(nodes, node)
+			maxBytes -= len(node) + io.GetVarSize(len(node))
+			return false
+		}
+		err := tr.Traverse(stop)
+		require.NoError(t, err)
+
+		var size int
+		for _, n := range nodes {
+			size += len(n) + io.GetVarSize(len(n))
+		}
+		require.Equal(t, maxSize, size+maxBytes)
+		require.Equal(t, expectedRoot, tr.StateRoot())
+	})
+
+	t.Run("Good, no restrictions", func(t *testing.T) {
+		var nodes [][]byte
+		stop := func(node []byte) bool {
+			nodes = append(nodes, node)
```

Comment:
How do we use `nodes` in this test?

Reply:
That's a remnant from the previous test, fixed.